src/Kind/Parse.hs

```diff
@@ -680,6 +681,13 @@ parsePatternNat = do
   let n = read num
   return $ (foldr (\_ acc -> PCtr "Succ" [acc]) (PCtr "Zero" []) [1..n])
 
+parsePatternLst :: Parser Pattern
+parsePatternLst = do
+  P.try $ char_skp '['
+  elems <- P.many parsePattern
```

Comment:
are lists comma or space separated? I can't quite tell, it looks implicit in this code.

Reply:
Space separated. It looks implicit because whitespaces are irrelevant (assuming we're using the right functions like `char_skp`, if we use Parsec directly then we need to manually skip).